### DISCUSSION TITLE
How to use Github Actions to Deploy multiple branch-specific Preview environments?

### DISCUSSION THREAD

**Discussion by dakdevs** [vercel/vercel#9141] [Help] [answered]
Hello 👋🏼 !

I am working on getting PR environments working with Vercel + Github Actions + PlanetScale. So far Vercel has roadblocked me a lot on this.

**What I am trying to achieve:**
- When a PR is opened/reopened a PlanetScale DB branch is created in Github Actions using the PlanetScale CLI.
- The `DATABASE_URL` is injected into the Vercel environment variables as a branch-specific Preview variable.
- On Github Actions, the NextJS app is built using the Preview environment variables with branch-specific environment variables overriding any base ones.
- The NextJS app is deployed to a Preview environment with runtime Preview and branch-specific environment variables.
- When a PR is closed or merged, the preview environment is torn down, the branch-specific environment variables are deleted, and the PlanetScale DB branch is either deleted or merged into the production branch.

When deploying with only Vercel you get **branch-specific deployments that utilize branch-specific preview environment variables**.

When deploying using the Vercel CLI, it seems to be impossible to achieve this. **Is this intended to force us to jam everything into our `vercel-build` command and not use any external CI/CD?**

How can I achieve this with Vercel?

**Comment by amyegan** [comment, marked as answer]
Sorry for the delay @Vinlock! You can specify the git branch metadata by using `vercel deploy -m githubCommitRef=my-branch`. That will let the deployment pick up the correct branch-specific preview environment variables.
